### PR TITLE
update Makefile for macosX installation about using luajit building deps

### PR DESCRIPTION
**Pull request by idevz**

### CODE REVIEW

**Comment by idevz** [comment]
I will add a macosx .travis soon to check this.

File: Makefile [hunk 1]
```diff
@@ -52,13 +54,13 @@ init:
 run:
 	mkdir -p logs
 	mkdir -p /tmp/cores/
-	$$(which openresty) -p $$PWD/
+	$(OR_EXEC) -p $$PWD/
```
Comment (membphis) [inline Makefile]: we need to specified fully path for config file, eg: `-c $$PWD/conf/nginx.conf`.

image

Comment (membphis) [reply Makefile]: it may fail under Mac OSX.

Comment (idevz) [reply Makefile]: yes, `-c` is needed under Mac OSX. I have add a travis for it, and all check passed. Let's merge it alright?

**Comment by membphis** [comment]
https://travis-ci.org/iresty/apisix/jobs/555975598

There are a lot of error logs on travis, please fix them at first.

File: utils/travis_runner.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        export OPENRESTY_PREFIX="/usr/local/openresty-debug"
+    fi
+}
+
+before_install() {
+    sudo cpanm --notest Test::Nginx IPC::Run >build.log 2>&1 || (cat build.log && exit 1)
+}
+
+do_install() {
+    if [ $TRAVIS_OS_NAME = osx ]; then
```
Comment (membphis) [inline utils/travis_runner.sh]: I think it's more convenient for different systems to use separate scripts.

how about: `***_mac.sh`, `***_linux.sh` ?

File: Makefile [hunk 3]
```diff
@@ -4,6 +4,8 @@ INST_LUADIR ?= $(INST_PREFIX)/share/lua/5.1
 INST_BINDIR ?= /usr/bin
 INSTALL ?= install
 UNAME ?= $(shell uname)
+OR_EXEC ?= $(shell which openresty)
+LUA_JIT_DIR ?= $(shell TMP='./v_tmp' && $(OR_EXEC) -V &>$${TMP} && cat $${TMP} | grep prefix | grep -Eo 'prefix=(.*?)/nginx' | grep -Eo '/.*/' && rm $${TMP})luajit
```
Comment (membphis) [inline Makefile]: './v_tmp' -> '/tmp/v_tmp', how about this?

File: utils/travis_runner.sh (hunk 2 above)
Comment (idevz) [reply utils/travis_runner.sh]: if using like `***_mac.sh`, `***_linux.sh` things, the control logic about to using which one will need to write in .travis.yaml, I think it's so ugly.

File: Makefile (hunk 3 above)
Comment (idevz) [reply Makefile]: it doesn't make any sense, the `v_tmp` file will been clean immediately at the end of this command.

File: utils/travis_runner.sh (hunk 2 above)
Comment (membphis) [reply utils/travis_runner.sh]: how about this way in travis: `bash utils/travis_$TRAVIS_OS_NAME.sh`

File: utils/travis_runner.sh [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        sudo luarocks install --lua-dir=${OPENRESTY_PREFIX}luajit apisix-*.rockspec --only-deps
+    fi
+
+    git clone https://github.com/openresty/test-nginx.git test-nginx
+}
+
+script() {
+    export_or_prefix
+    export PATH=$OPENRESTY_PREFIX/nginx/sbin:$OPENRESTY_PREFIX/luajit/bin:$OPENRESTY_PREFIX/bin:$PATH
+    if [ $TRAVIS_OS_NAME = osx ]; then
```
Comment (membphis) [inline utils/travis_runner.sh]: I prefer this style:

```
# mac os
if [ $TRAVIS_OS_NAME = osx ]; then
   ...
   return
fi

# default linux
...
```

**Comment by membphis** [comment]
Code coverage check seems to have failed, but I can not sure for this. @moonming pls take a look at this.

image

**Pull request merged**